lib/shopify_app/controller_concerns/token_exchange.rb

```diff
@@ -83,8 +89,36 @@ def redirect_to_bounce_page
       )
     end
 
+    def missing_embedded_param?
+      !params[:embedded].present? || params[:embedded] != "1"
+    end
+
+    def redirect_to_embed_app
```

Comment:
Just out of curiosity - won't we have other places in the code that need to do this as well? Seems to me like this might also be extractable if that's the case.